### PR TITLE
[#2340] improvement(api/catalog/spark/trino): support unsigned type

### PR DESCRIPTION
**Pull request by hanwxx**
### What changes were proposed in this pull request?

support mysql unsigned integer type

### Why are the changes needed?
Fix: #2340

### Does this PR introduce _any_ user-facing change?
N/A

### How was this patch tested?
UT

### CODE REVIEW

**Comment by YxAc** [comment]
could you please comply with the PR specification, i.e., title like #3315 , which including all issues fixed and specifying ‘improvement’

**Comment by hanwxx** [comment]
> could you please comply with the PR specification, i.e., title like #3315 , which including all issues fixed and specifying ‘improvement’ module.

thanks, i will modify the title

**Comment by mchades** [comment]
Please add comments to the public methods in the API module to make the tests pass

**Comment by hanwxx** [comment]
> Please add comments to the public methods in the API module to make the tests pass

thanks, i will add the comment

File: api/src/main/java/com/datastrato/gravitino/rel/types/Type.java [hunk 1]
```diff
@@ -98,7 +98,19 @@ abstract class IntervalType extends PrimitiveType {}
   abstract class ComplexType implements Type {}
 
   /** The base type of all integral types. */
-  abstract class IntegralType extends NumericType {}
+  abstract class IntegralType extends NumericType {
+    private final boolean signed;
```
Comment (yuqi1129) [inline api/src/main/java/com/datastrato/gravitino/rel/types/Type.java:102]: I would suggest rename it to `unsigned` as data type like `integer` is signed by default, and the default value of bool type is `false`.

Comment (mchades) [reply api/src/main/java/com/datastrato/gravitino/rel/types/Type.java:102]: The current implementation is LGTM. `signed` is in the constructor and is more intuitive

File: spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTypeConverter.java
Comment (mchades) [inline spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTypeConverter.java:1]: @FANNG1 Could you please help to review the changes to this part?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/util/GeneralDataTypeTransformer.java
Comment (mchades) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/util/GeneralDataTypeTransformer.java:1]: @yuqi1129 @diqiu50 Could you please help to review the changes to Trino connector part?

**Comment by mchades** [comment]
- The `unsignedXxxLiteral(xxx)` method also needs to be added in `com.datastrato.gravitino.rel.expressions.literals.Literals`
 - The docs need to be modified: 
   - `docs/open-api/datatype.yaml`
   - `docs/manage-relational-metadata-using-gravitino.md:695`
   - `docs/jdbc-mysql-catalog.md:88`

**Comment by hanwxx** [comment]
> * The `unsignedXxxLiteral(xxx)` method also needs to be added in `com.datastrato.gravitino.rel.expressions.literals.Literals`
> * The docs need to be modified:
>   
>   * `docs/open-api/datatype.yaml`
>   * `docs/manage-relational-metadata-using-gravitino.md:695`
>   * `docs/jdbc-mysql-catalog.md:88`
get, i will add it

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/util/GeneralDataTypeTransformer.java
Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/util/GeneralDataTypeTransformer.java:1]: It's okay for me in this aspect.

File: spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTypeConverter.java [hunk 2]
```diff
@@ -100,13 +100,29 @@ public static Type toGravitinoType(DataType sparkType) {
 
   public static DataType toSparkType(Type gravitinoType) {
     if (gravitinoType instanceof Types.ByteType) {
-      return DataTypes.ByteType;
+      if (((Types.ByteType) gravitinoType).signed()) {
```
Comment (FANNG1) [inline spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTypeConverter.java:103]: Spark SQL doesn't support unsigned types,  it's strange to change like this.

Comment (FANNG1) [reply spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTypeConverter.java:103]: I prefer to throw exceptions like unsigned types for Spark

**Comment by yuqi1129** [comment]
@hanwxx 
Could you please resolve the conflict if you have time?

**Comment by yuqi1129** [comment]
@mchades 
Can you please help to review this RP and move forward with it?

**Comment by hanwxx** [comment]
> @hanwxx Could you please resolve the conflict if you have time?
get, has resolve

**Comment by yuqi1129** [comment]
> LGTM, @yuqi1129 @FANNG1 do you want to take another look?

I have no further comments on the matter.

**Pull request merged**